# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIR.h
# Commit: 229759a67f4f
# Full Hash: 229759a67f4f26ccde9f7bde5423cfd82b216fa2
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-03-22 01:23:00
# Description:
#   Bug 1537924 - Simplify some alias sets in Ion. r=tcampbell, a=dveditz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D24400
# ==============================================================================

diff -r feda786b35cb -r 229759a67f4f js/src/jit/MIR.h
--- a/js/src/jit/MIR.h	Thu Mar 21 23:59:53 2019 +0200
+++ b/js/src/jit/MIR.h	Thu Mar 21 17:52:21 2019 +0000
@@ -8082,9 +8082,6 @@
 
   gc::InitialHeap initialHeap() const { return initialHeap_; }
 
-  AliasSet getAliasSet() const override {
-    return AliasSet::Store(AliasSet::Element | AliasSet::ObjectFields);
-  }
   bool possiblyCalls() const override { return true; }
   bool appendRoots(MRootList& roots) const override {
     return roots.append(templateObj_);
@@ -9665,7 +9662,8 @@
     if (!idempotent_) {
       return AliasSet::Store(AliasSet::Any);
     }
-    return AliasSet::None();
+    return AliasSet::Load(AliasSet::ObjectFields | AliasSet::FixedSlot |
+                          AliasSet::DynamicSlot);
   }
   bool possiblyCalls() const override { return true; }
   bool appendRoots(MRootList& roots) const override {
